src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java

```diff
@@ -1,11 +1,28 @@
 package frc.robot.subsystems.drive;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-import frc.robot.Constants.DriveMode;
 import frc.robot.Constants.ChassisMode;
 import frc.robot.Constants.DriveConstants;
+import frc.robot.Constants.DriveMode;
+import frc.robot.RobotState;
 
 public class DriveSubsystem extends SubsystemBase{
 
+
+    private final SwerveModules[] driveModules = new SwerveModules[4];
+    private final DriveAngle driveAngle = new DriveAngle();
+    private SwerveModulePosition modulePositions[] = new SwerveModulePosition[4];
+    private SwerveModuleState[] goalModuleStates = new SwerveModuleState[4];
+    private PIDController[] rotationPIDs = new PIDController[4];
+
     private DriveMode mode;
 
     private DriveHardware driveHardware;
```

Comment:
Let's get rid of drive hardware related things from DriveSubsystem for now

Reply:
done